components/server/ee/src/workspace/gitpod-server-impl.ts

```diff
@@ -2057,6 +2068,34 @@ export class GitpodServerEEImpl extends GitpodServerImpl {
         }
     }
 
+    protected defaultSpendingLimit = 100;
+    async getSpendingLimitForTeam(ctx: TraceContext, teamId: string): Promise<number | undefined> {
+        const user = this.checkAndBlockUser("getSpendingLimitForTeam");
+        await this.ensureIsUsageBasedFeatureFlagEnabled(user);
+        await this.guardTeamOperation(teamId, "get");
+        const attributionId = AttributionId.render({ kind: "team", teamId });
+        const costCenter = await this.costCenterDB.findById(attributionId);
+        if (costCenter) {
+            return costCenter.spendingLimit;
+        }
+        // On demand creating a cost center for this team
+        await this.costCenterDB.storeEntry({
+            id: attributionId,
+            spendingLimit: this.defaultSpendingLimit,
+        });
+        return this.defaultSpendingLimit;
+    }
+
+    async setSpendingLimitForTeam(ctx: TraceContext, teamId: string, spendingLimit: number): Promise<void> {
+        const user = this.checkAndBlockUser("setSpendingLimitForTeam");
+        await this.ensureIsUsageBasedFeatureFlagEnabled(user);
+        await this.guardTeamOperation(teamId, "update");
+        await this.costCenterDB.storeEntry({
```

Comment:
We should add validation here. It's unlikely negative `spendingLimit` is a valid value.

Reply:
good point. updated